### DISCUSSION TITLE
How do you use the right prompt?

### DISCUSSION THREAD

**Discussion by floam** [fish-shell/fish-shell#8668] [Show and tell]
I'm curious how people have set up their right prompts. I used to report `$CMD_DURATION`. More recently, I've found it useful to keep track of my laptop's battery. It reminds me to plug it in.

<img width="381" alt="Screen Shot 2022-01-28 at 2 24 18 AM" src="https://user-images.githubusercontent.com/291142/151530572-1dcde7b3-8515-4095-9f8a-df3e4651a989.png">

```fish
# Defined in /Users/floam/.config/fish/functions/fish_right_prompt.fish @ line 1
function fish_right_prompt
    /usr/bin/pmset -g batt | string match -gar '([0-9]+\%); (.*charging)' | read -lL percent state
    [ "$_last_percent" = "$percent" -a "$_last_state" = "$state" ]
    and return
    set_color -r -o
    switch $state
        case charging
            set_color brgreen
        case discharging
            if [ "$(string replace -ra % '' $percent)" -le 15 ]
                set_color brred
            else
                set_color bryellow
            end
    end
    printf %s $percent (set_color normal)
    set -g _last_percent $percent
    set -g _last_state $state
    set_color normal
end
```

**Comment by Darkshadow2** [comment]
I use the right prompt to show the time and git status:

![Screen Shot 2022-02-13 at 6 30 00 PM](https://user-images.githubusercontent.com/896142/153780196-45079ef8-c1b6-4470-8490-447ec09fdc23.png)

I also have different colors for different users (on various different systems). Just a visual clue to where I am / who I am.

My prompt is pretty nontrivial, but here's just that portion:

```fish
# Defined in /fish/git/functions/fish_right_prompt.fish @ line 2
function fish_right_prompt
	__ds_powerline_right
end
```

```fish
# Defined in /fish/git/conf.d/powerline.fish @ line 172
function __ds_powerline_right
	if not set -q __ds_powerline_right_segments[1]
		return 0
	end
	set -l segment
	__ds_powerline_setup_vars false
	if set -q __ds_powerline_highlight_color[1] && set -q __ds_powerline_drawing_multiline[1]
		set __ds_color_normal $__ds_powerline_highlight_color
	end
	if set -q __ds_powerline_drawing_multiline
		echo -n $__ds_powerline_tput_sc$__ds_powerline_tput_cuu1$__ds_powerline_tput_cuf2
	end
	for segment in $__ds_powerline_right_segments
		if test -z $segment; continue; end
		set -l seg_func __ds_powerline_{$segment}_draw
		if not functions -q $seg_func
			__ds_powerline_draw_whole_segment "prompt_error" "$segment not a valid segment"
		else
			$seg_func 2>/dev/null
		end
	end
	set_color normal
	if set -q __ds_powerline_drawing_multiline
		echo -n $__ds_powerline_tput_rc
		set -e __ds_powerline_drawing_multiline
	end
	__ds_powerline_teardown_vars
end
```

My prompt isn't actually multiline, the actual prompt is only that second line.  The first is output when `fish_prompt` is emitted, so that's why my right prompt has cursor movement shenanigans.

**Reply by spenserblack** [reply]
I have to say that that ASCII art is really cool.

**Reply by Darkshadow2** [reply]
Indeed it is!  It's not mine, it's by Jeff Ferris, found [here](https://asciiart.website//index.php?art=creatures/dragons) (scroll down a bit). I modified it slightly (simplified it somewhat). That ascii art site is great.

**Comment by ipatch** [comment]
i have the right prompt mostly disabled, but have a keybinding where i can toggle it on or off. it's pretty hacky to say the least and requires creating a file that stores the display status of the right prompt. ie. a little state file. i found it was getting more so in the way, when i would scroll back up and have to copy large blocks of text, thus i mostly have it hidden / disabled, but do enable it from time to time.
